### PR TITLE
Paper Recruitment Pre Process Function

### PR DESCRIPTION
**Pull request by purujitgoyal**

### CODE REVIEW

File: openreview/conference/templates/paper_recruitment_pre_process.py [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+    submission = client.get_notes(note.content['submission_id'], details='original')[0]
+
+    if hashkey != note.content['key']:
+        raise openreview.OpenReviewException('Wrong key, please refer back to the recruitment email')
+
+    if not client.get_groups(id=REVIEWERS_INVITED_ID, member=user):
+        raise openreview.OpenReviewException('User not in invited group, please accept the invitation using the email address you were invited with')
+
+    if 'Desk_Rejected_Submission' in submission.invitation or 'Withdrawn_Submission' in submission.invitation:
+        raise openreview.OpenReviewException('This submission is no longer under consideration. Please contact the paper area chair or program chairs to be unassigned.')
```
Comment (melisabok) [inline openreview/conference/templates/paper_recruitment_pre_process.py:21]: @celestemartinez could you help here with the error message? The user that accept or declines an invitation to review a paper that is not under review anymore.

**Comment by melisabok** [comment]
Looks good! Waiting for Celeste's feedback.

File: tests/test_arr_venue.py [hunk 2]
```diff
@@ -756,3 +756,45 @@ def test_setup_matching(self, venue, client, helpers, request_page, selenium):
...
+            }
+        )
+
+        posted_note = pc_client.post_note(desk_reject_note)
+        assert posted_note
+
+        helpers.await_queue()
+
+        ## Reviewer reviewer_arr2@mit.edu tries to accept invitation for desk-rejected submission but gets an error
+        invite_edges = pc_client.get_edges(invitation='aclweb.org/ACL/ARR/2021/September/Reviewers/-/Invite_Assignment',
```
Comment (melisabok) [inline tests/test_arr_venue.py:783]: it is strange that edge can be posted if the note is already desk rejected.

Comment (purujitgoyal) [reply tests/test_arr_venue.py:783]: edge was already posted before the desk-reject, it's just getting the posted edge here.

File: openreview/conference/templates/paper_recruitment_pre_process.py (hunk 1 above)
Comment (celestemartinez) [reply openreview/conference/templates/paper_recruitment_pre_process.py:21]: What about
> This submission is no longer under review. No action is required from your end. 

or something similar? I don't think the AC/PCs should be contacted to unassign them (if I am not mistaken, the edge browser won't even show this paper anymore).

File: tests/test_arr_venue.py (hunk 2 above)
Comment (melisabok) [reply tests/test_arr_venue.py:783]: You are right 🤦‍♀️

File: openreview/conference/templates/paper_recruitment_pre_process.py (hunk 1 above)
Comment (melisabok) [reply openreview/conference/templates/paper_recruitment_pre_process.py:21]: That's true, the user should ignore the invitation.

**Review by melisabok — APPROVED** [review]

**Comment by melisabok** [comment]
@celestemartinez can I merge this one?

**Review by celestemartinez — APPROVED** [review]

**Pull request merged**